kyo-prelude/shared/src/main/scala/kyo/Memo.scala

```diff
@@ -58,39 +58,11 @@ object Memo:
     def run[A: Flat, S](v: A < (Memo & S))(using Frame): A < S =
         Var.run(empty)(v)
 
-    object isolate:
-
-        /** Creates an isolate that combines memoization caches.
-          *
-          * When the isolation ends, merges any cached results from the isolated computation with the outer cache using the later result on
-          * conflicts. This allows memoized results computed in isolation to be reused later.
-          *
-          * @return
-          *   An isolate that preserves memoized results
-          */
-        val merge: Isolate[Memo] =
-            Var.isolate.merge[Memo.Cache]((m1, m2) => Cache(m1.map ++ m2.map))
-
-        /** Creates an isolate that overwrites the memoization cache.
-          *
-          * When the isolation ends, replaces the outer cache with the cache from the isolated computation. Earlier cached results are
-          * discarded in favor of any new results computed in isolation.
-          *
-          * @return
-          *   An isolate that updates the cache with isolated results
-          */
-        val update: Isolate[Memo] =
-            Var.isolate.update[Memo.Cache]
+    /** Default isolate that combines memoization caches.
+      *
+      * When the isolation ends, merges any cached results from the isolated computation with the outer cache using the later result on
+      * conflicts. This allows memoized results computed in isolation to be reused later.
+      */
+    given isolate: Isolate.Stateful[Memo, Any] = Var.isolate.merge[Cache]((a, b) => Cache(a.map ++ b.map))
```

Comment:
I've also decided to provide an evidence for `Memo` that always merges results. The fine-grained isolation options didn't seem very valuable.